tests/unit/nupic/research/temporal_memory_test.py

```diff
@@ -578,6 +585,73 @@ def testMapCellsToColumns(self):
     self.assertEqual(columnsForCells[99], set([399]))
 
 
+  def testWrite(self):
+    tm1 = TemporalMemory(
+      columnDimensions=[100],
+      cellsPerColumn=4,
+      activationThreshold=7,
+      initialPermanence=0.37,
+      connectedPermanence=0.58,
+      minThreshold=4,
+      maxNewSynapseCount=18,
+      permanenceIncrement=0.23,
+      permanenceDecrement=0.08,
+      seed=91
+    )
+
+    # Run some data through before serializing
+    self.patternMachine = PatternMachine(100, 4)
+    self.sequenceMachine = SequenceMachine(self.patternMachine)
+    sequence = self.sequenceMachine.generateFromNumbers(range(5))
+    for _ in range(3):
+      for pattern in sequence:
+        tm1.compute(pattern)
+
+    proto1 = TemporalMemoryProto_capnp.TemporalMemoryProto.new_message()
+    tm1.write(proto1)
+
+    # Write the proto to a temp file and read it back into a new proto
+    with tempfile.TemporaryFile() as f:
+      proto1.write(f)
+      f.seek(0)
+      proto2 = TemporalMemoryProto_capnp.TemporalMemoryProto.read(f)
+
+    # Load the deserialized proto
+    tm2 = TemporalMemory.read(proto2)
+
+    # Check that the two temporal memory objects have the same attributes
+    self.assertEqual(tm1.columnDimensions, tm2.columnDimensions)
```

Comment:
@chetan51 - do you think it would be better to define a `__eq__` method on TemporalMemory that makes these checks?